File: app/src/main/java/com/example/music/ui/adapters/HomeGridAdapter.java
package com.example.music.ui.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.music.R;
import com.example.music.data.model.Track;

import java.util.List;

public class HomeGridAdapter extends RecyclerView.Adapter<HomeGridAdapter.ViewHolder> {

    private final List<Track> items;
    private final OnItemClickListener listener;

    public interface OnItemClickListener {
        void onItemClick(Track track);
    }

    public HomeGridAdapter(List<Track> items, OnItemClickListener listener) {
        this.items = items;
        this.listener = listener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_recent_mini, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Track item = items.get(position);
        holder.title.setText(item.getTitle());
        // Load image using Glide or similar if available, for now just placeholder
        // logic
        // holder.image.setImageResource(item.getImageResId());

        holder.itemView.setOnClickListener(v -> {
            if (listener != null) {
                listener.onItemClick(item);
            }
        });
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        ImageView image;
        TextView title;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            image = itemView.findViewById(R.id.image_mini);
            title = itemView.findViewById(R.id.text_mini_title);
        }
    }
}

File: app/src/main/java/com/example/music/ui/adapters/HomeGridAdapter.kt
package com.example.music.ui.adapters

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.music.R
import com.example.music.data.model.Track
import com.example.music.databinding.ItemRecentMiniBinding

class HomeGridAdapter(
    private val tracks: List<Track>,
    private val onClick: (Track) -> Unit
) : RecyclerView.Adapter<HomeGridAdapter.ViewHolder>() {

    inner class ViewHolder(private val binding: ItemRecentMiniBinding) :
        RecyclerView.ViewHolder(binding.root) {

        fun bind(track: Track) {
            binding.textMiniTitle.text = track.title
            Glide.with(binding.root.context)
                .load(track.album.coverMedium)
                .placeholder(R.color.spotify_light_grey)
                .centerCrop()
                .into(binding.imageMini)
            binding.root.setOnClickListener { onClick(track) }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding = ItemRecentMiniBinding.inflate(
            LayoutInflater.from(parent.context), parent, false
        )
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) =
        holder.bind(tracks[position])

    override fun getItemCount() = tracks.size
}
